packages/integration-sdk-runtime/src/execution/dependencyGraph.ts

```diff
@@ -154,6 +154,11 @@ export function executeStepDependencyGraph<
         ...existingResult,
         status,
         encounteredTypes: typeTracker.getEncounteredTypesForStep(stepId),
```

Comment:
Not modifying the existing `encounteredTypes` since it is used elsewhere and would be a breaking change.